packages/core/src/components/button/buttons.tsx

```diff
@@ -13,20 +13,26 @@ import * as React from "react";
 
 import * as Classes from "../../common/classes";
 import { IActionProps, removeNonHTMLProps } from "../../common/props";
+import { Spinner } from "../spinner/spinner";
 
 export interface IButtonProps extends IActionProps {
     /** A ref handler that receives the native HTML element backing this component. */
     elementRef?: (ref: HTMLElement) => any;
 
     /** Name of icon (the part after `pt-icon-`) to add to button. */
     rightIconName?: string;
+
+    /** Whether the button's action is currently loading (will disable the button) */
```

Comment:
not the best explanation of what the prop does. i like what you have in the PR description about showing a spinner but preserving the button width.